gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDatasetFinder.java

```diff
@@ -92,6 +103,18 @@ public Iterator<IcebergDataset> getDatasetsIterator() throws IOException {
 
   protected IcebergDataset createIcebergDataset(String dbName, String tblName, IcebergCatalog icebergCatalog, Properties properties, FileSystem fs) {
     IcebergTable icebergTable = icebergCatalog.openTable(dbName, tblName);
-    return new IcebergDataset(dbName, tblName, icebergTable, properties, fs);
+    return new IcebergDataset(dbName, tblName, icebergTable, icebergCatalog.getCatalogUri(), properties, fs);
```

Comment:
seems worthwhile to keep around the `IcebergCatalog.getCatalogUri()` method, but rather than calling here, why not have that catalog pass its URI to the `IcebergTable`, as the former is the very one to create the latter?

subsequently `IcebergDataset` will call `IcebergTable.getDatasetDescriptor`, w/o needing to pass in a URI arg.

Reply:
makes sense... I have associated the catalog Uri with `IcebergTable` in my latest commit!